network/transport/protocol.go

```diff
@@ -36,4 +36,20 @@ type Protocol interface {
 	PeerDiagnostics() map[PeerID]Diagnostics
 	// Version returns the version of the protocol
 	Version() int
+	// MethodName returns the fully qualified name of the gRPC stream.
```

Comment:
gRPC detail outside of gRPC package. Is it here to fix cyclic dependencies?

Reply:
moved back to `grpc.Protocol`